### PR TITLE
Add sessions-, project- and groups- ACL checks. Fix files upload code

### PR DESCRIPTION
**Pull request by haakonvt**
~~This PR awaits the merging of #3 ~~

## This is the first proper version of the OIDC function-action. Enjoy ❤️

### CODE REVIEW

File: src/access.py [hunk 1]
```diff
@@ -18,10 +18,11 @@ def verify_credentials_vs_project(client: CogniteClient, project: str, cred_name
     try:
         token_inspect = client.iam.token.inspect()
     except CogniteAPIError:
-        raise ValueError(
-            f"{cred_name.title()} credentials wrong or missing one or more capabilities! "
-            "Requires both 'Projects:LIST' and 'Groups:LIST'!"
+        logger.error(
+            f"Unable to call 'token/inspect'. Probable cause: {cred_name.title()} credentials wrong or missing "
+            "one or more capabilities! Requires both 'Projects:LIST' and 'Groups:LIST'!"
         )
```
Comment (haakonvt) [inline src/access.py:24]: Note to self: What happens if you only lack `Groups:LIST`?

File: src/access.py [hunk 2]
```diff
@@ -77,37 +89,48 @@ def missing_files_capabilities(
...
 
 
-def verify_capabilites(
+def verify_schedule_creds_capabilities(token_inspect: TokenInspection, project: str) -> None:
+    capabilities = parse_capabilities_from_token_inspect_for_project(token_inspect, project)
+    missing = missing_session_capabilities(capabilities)
+    if missing:
+        err_msg = f"Schedule credentials missing one or more required capabilities:\n{linesep.join(missing)}"
+        logger.error(err_msg)
+        raise ValueError(err_msg)
```
Comment (haakonvt) [inline src/access.py]: Note to self: can separate out with `NoReturn`

**Review by pavelzubarev — COMMENTED** [review]
Are you sure you don't want to treat token as an opaque token? Parsing tokens is an antipattern.

File: src/function_file.py [hunk 3]
```diff
@@ -22,19 +23,28 @@ def _write_files_to_zip_buffer(zf: ZipFile, directory: Path):
             zf.write(Path(dirpath) / f)
 
 
+@retry(exceptions=FunctionDeployError, tries=12, delay=2, jitter=2, max_delay=15, logger=None)
```
Comment (pavelzubarev) [inline src/function_file.py:26]: holly ...!

**Review by pavelzubarev — APPROVED** [review]
I think we can move forward with this. I want to point out that parsing tokens is not a good idea. That format and content may change anytime, that will kill your code for no reason, most likely. (I do understand you want to prevent known pitfalls, but I have to point that out)

File: src/function_file.py (hunk 3 above)
Comment (haakonvt) [reply src/function_file.py:26]: Haha, just waiting for a file being uploaded 😉

**Comment by haakonvt** [comment]
> I think we can move forward with this. I want to point out that parsing tokens is not a good idea. That format and content may change anytime, that will kill your code for no reason, most likely. (I do understand you want to prevent known pitfalls, but I have to point that out)

Should I just change to using `client.iam.groups.list()` instead?

**Pull request merged**